File: single-language-analysis/dataset-files/java/TestSmell122.java
import org.junit.Test;
import static org.junit.Assert.*;
import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class TestSmell122 {

    @Test
    public void testThreadSafety() {
        List<String> errors = new ArrayList<>();

        Runnable cb = new Runnable() {
            @Override
            public void run() {
                try {
                    Thread.sleep(1e-3);
                    int r = module.t(() -> 123);
                    assertEquals(123, r);
                } catch (InterruptedException e) {
                    Thread.currentThread().interrupt();
                }
            }
        };

        Runnable runner = new Runnable() {
            @Override
            public void run() {
                try {
                    for (int j = 0; j < 50; j++) {
                        int r = module.t(cb);
                        assertEquals(42, r);
                        checkFunction("t");
                    }
                } catch (Exception e) {
                    errors.add(e.toString());
                }
            }
        };

        ExecutorService executor = Executors.newFixedThreadPool(20);
        for (String arg : new String[]{"t", "t2"}) {
            for (int n = 0; n < 20; n++) {
                executor.submit(runner);
            }
        }
        executor.shutdown();
        while (!executor.isTerminated()) {}

        String errorMessages = String.join("\n\n", errors);
        if (!errorMessages.isEmpty()) {
            throw new AssertionError(errorMessages);
        }
    }
}

File: single-language-analysis/dataset-files/CSharp/CSharp.Tests/Tests/TestSmell122.cs
using Xunit;

public class TestSmell122
{
    [Fact]
    public void TestThreadSafety()
    {
        var errors = new List<string>();

        void Cb()
        {
            System.Threading.Thread.Sleep(1e-3);
            var r = this.Module.T(() => 123);
            Assert.Equal(123, r);
            return 42;
        }

        void Runner(string name)
        {
            try
            {
                for (int j = 0; j < 50; j++)
                {
                    var r = this.Module.T(Cb);
                    Assert.Equal(42, r);
                    CheckFunction(name);
                }
            }
            catch (Exception)
            {
                errors.Add(Exception.ToString());
            }
        }

        var threads = new List<System.Threading.Thread>();
        foreach (var arg in new[] { "t", "t2" })
        {
            for (int n = 0; n < 20; n++)
            {
                threads.Add(new System.Threading.Thread(() => Runner(arg)));
            }
        }

        foreach (var t in threads)
        {
            t.Start();
        }

        foreach (var t in threads)
        {
            t.Join();
        }

        var errorMessages = string.Join("\n\n", errors);
        if (!string.IsNullOrEmpty(errorMessages))
        {
            throw new Xunit.Sdk.AssertActualExpectedException(errorMessages, null, null);
        }
    }
}